frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx

```diff
@@ -58,6 +72,32 @@ const UserProfileSidebar: React.FC<Properties> = ({
     void dispatch(authActions.signOut());
   }, [dispatch]);
 
+  const handleClose = useCallback((): void => {
+    dialogReference.current?.open && dialogReference.current.close();
+  }, [dialogReference]);
+
+  const handleOpen = useCallback(() => {
+    setCurrentStepIndex(INITIAL_STEP);
+    dialogReference.current?.showModal();
+  }, [dialogReference]);
+
+  const goToNextStep = useCallback((): void => {
+    setCurrentStepIndex((previous) => {
+      return Math.min(
+        previous + NEXT_STEP_INCREMENT,
+        STEPS.length - NEXT_STEP_INCREMENT,
+      );
+    });
+  }, []);
+
+  const getCurrentStep = (): ReactElement | null => {
+    const StepComponent = currentStep?.component;
```

Comment:
You are checking Boolean(currentStep) before calling this function. Use type assertion for this case